### PR TITLE
fix: [OIP-870]: Fix for breaking UI in monitored service templates

### PR DESCRIPTION
**Pull request by praneshg239**
#### Screenshots

Pasted Graphic 3
Pasted Graphic 4
type MonitoredService

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/85-cv/pages/monitored-service/components/Configurations/Configurations.tsx [hunk 1]
```diff
@@ -189,11 +189,14 @@ export default function Configurations(
...
   useEffect(() => {
     const cloneTemplateValue = clone(templateValue)
+
     if (isTemplate && cloneTemplateValue?.name && cloneTemplateValue?.spec?.sources && storeMetadata?.storeType) {
       if (!identifier && isEmpty(cloneTemplateValue?.spec?.sources?.changeSources) && !hasTemplateChangeSourceSet) {
         cloneTemplateValue.spec.sources['changeSources'] = defaultMonitoredService?.sources?.changeSources
         sethasTemplateChangeSourceSet(true)
       }
+
+      // This is causing the reset
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/monitored-service/components/Configurations/Configurations.tsx:199]: we can remove this comment

File: src/modules/85-cv/pages/monitored-service/components/Configurations/Configurations.tsx [hunk 2]
```diff
@@ -232,7 +235,14 @@ export default function Configurations(
     if (identifier && fetchMonitoredService) {
       fetchMonitoredService()
     } else if ((isTemplate && isNewTemplate(templateIdentifier)) || (!isTemplate && !identifier)) {
-      fetchMonitoredServiceYAML()
+      fetchMonitoredServiceYAML({
+        queryParams: {
+          accountId,
+          orgIdentifier,
+          projectIdentifier,
+          type: templateValue?.spec?.type
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/monitored-service/components/Configurations/Configurations.tsx:243]: if templateValue?.spec?.type  is undefined or null , let's not send type at all

File: src/modules/85-cv/pages/monitored-service/components/Configurations/__tests__/Configurations.test.tsx [hunk 3]
```diff
@@ -104,7 +103,8 @@ mockImport('framework/LicenseStore/LicenseStoreContext', {
 })
 
 describe('Unit tests for Configuration', () => {
-  test('Ensure that any infra change source is removed when switching type to application', async () => {
+  // eslint-disable-next-line jest/no-disabled-tests
+  test.skip('Ensure that any infra change source is removed when switching type to application', async () => {
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/monitored-service/components/Configurations/__tests__/Configurations.test.tsx:107]: why we skipping this ?

File: src/modules/85-cv/pages/monitored-service/components/Configurations/Configurations.tsx (hunk 1 above)
Comment (praneshg239) [reply src/modules/85-cv/pages/monitored-service/components/Configurations/Configurations.tsx]: Done

File: src/modules/85-cv/pages/monitored-service/components/Configurations/Configurations.tsx (hunk 2 above)
Comment (praneshg239) [reply src/modules/85-cv/pages/monitored-service/components/Configurations/Configurations.tsx]: Done

File: src/modules/85-cv/pages/monitored-service/components/Configurations/__tests__/Configurations.test.tsx (hunk 3 above)
Comment (praneshg239) [reply src/modules/85-cv/pages/monitored-service/components/Configurations/__tests__/Configurations.test.tsx]: Skipped for time basis, forgot to un-skip later. It is fixed now, please check.

**Review by deepesh-ui — APPROVED** [review]

**Review by karansaraswat19 — APPROVED** [review]

**Comment by sribalij** [comment]
@praneshg239 Can we merge this?

**Pull request merged**